art/detection/subsetscanning/detector.py

```diff
@@ -0,0 +1,184 @@
+""" Fast generalized subset scan based detector"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+import logging
+import six
+
+from sklearn import metrics
+import numpy as np
+
+from art.classifiers import Classifier
+from art.detection.subsetscanning.scanner import Scanner
+
+logger = logging.getLogger(__name__)
+
+
+class SubsetScanningDetector(Classifier):
+    """ Fast generalized subset scan based detector"""
+    def __init__(self, classifier, bgd_data, layer):
+        super(SubsetScanningDetector, self).__init__(clip_values=classifier.clip_values,
+                                                     channel_index=classifier.channel_index,
+                                                     defences=classifier.defences,
+                                                     preprocessing=classifier.preprocessing)
+        self.classifier = classifier
+        self.bgd_data = bgd_data
+
+        # Ensure that layer is well-defined:
+        if isinstance(layer, six.string_types):
+            if layer not in classifier.layer_names:
+                raise ValueError('Layer name %s is not part of the graph.' % layer)
+            self._layer_name = layer
+        elif isinstance(layer, int):
+            if layer < 0 or layer >= len(classifier.layer_names):
+                raise ValueError('Layer index %d is outside of range (0 to %d included).'
+                                 % (layer, len(classifier.layer_names) - 1))
+            self._layer_name = classifier.layer_names[layer]
+        else:
+            raise TypeError('Layer must be of type `str` or `int`.')
+
+        bgd_activations = classifier.get_activations(bgd_data, self._layer_name)
+        if len(bgd_activations.shape) == 4:
+            dim2 = bgd_activations.shape[1] * bgd_activations.shape[2] * bgd_activations.shape[3]
+            bgd_activations = np.reshape(bgd_activations, (bgd_activations.shape[0], dim2))
+
+        self.sorted_bgd_activations = np.sort(bgd_activations, axis=0)
+
+
+    def calculate_pvalue_ranges(self, x):
```

Comment:
Could you please add a docstring describing what the function does, its arguments and return similar to the other function docstrings in ART?